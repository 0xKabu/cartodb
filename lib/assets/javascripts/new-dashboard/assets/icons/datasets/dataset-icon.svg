<?xml version="1.0" encoding="UTF-8"?>
<svg width="36px" height="40px" viewBox="0 0 36 40" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 52.6 (67491) - http://www.bohemiancoding.com/sketch -->
    <title>server-2</title>
    <desc>Created with Sketch.</desc>
    <g id="Active-Development-Cycle" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="01-Home-/-Recent" transform="translate(-382.000000, -318.000000)">
            <g id="Group-6" transform="translate(376.000000, 312.000000)">
                <g id="places-home-1">
                    <g id="Frame_-_24px">
                        <g id="server-2" transform="translate(0.000000, 2.000000)">
                            <g id="Frame-24px">
                                <rect id="Rectangle" x="0" y="0" width="48" height="48"></rect>
                            </g>
                            <g id="Line_Icons" transform="translate(6.000000, 4.000000)" fill="#FFFFFF" fill-rule="nonzero">
                                <path d="M18,0 C9.054,0 0,2.748 0,8 L0,32 C0,37.252 9.054,40 18,40 C26.946,40 36,37.252 36,32 L36,8 C36,2.748 26.946,0 18,0 Z M18,4 C27.094,4 32,6.84 32,8 C32,9.16 27.094,12 18,12 C8.906,12 4,9.16 4,8 C4,6.84 8.906,4 18,4 Z M18,36 C8.906,36 4,33.16 4,32 L4,29.2 C7.472,31.046 12.748,32 18,32 C23.252,32 28.528,31.044 32,29.2 L32,32 C32,33.16 27.094,36 18,36 Z M18,28 C8.906,28 4,25.16 4,24 L4,21.2 C7.472,23.046 12.748,24 18,24 C23.252,24 28.528,23.044 32,21.2 L32,24 C32,25.16 27.094,28 18,28 Z M18,20 C8.906,20 4,17.16 4,16 L4,13.2 C7.472,15.044 12.748,16 18,16 C23.252,16 28.528,15.044 32,13.2 L32,16 C32,17.16 27.094,20 18,20 Z" id="Shape"></path>
                            </g>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>