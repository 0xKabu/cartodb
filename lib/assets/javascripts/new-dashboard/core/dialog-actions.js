// Dialogs
import Dialog from 'new-dashboard/components/Backbone/Dialog.vue';
import ChangePrivacy from 'new-dashboard/components/Backbone/Dialogs/ChangePrivacy';
import DuplicateMap from 'new-dashboard/components/Backbone/Dialogs/DuplicateMap';
import LockMap from 'new-dashboard/components/Backbone/Dialogs/LockMap';
import DeleteDialog from 'new-dashboard/components/Backbone/Dialogs/DeleteDialog';

export function changePrivacy (visualization) {
  this.showModal(
    {
      template: `
      <Dialog v-on:close="$emit('close')">
        <ChangePrivacy :visualization="visualization" v-on:close="$emit('close')"/>
      </Dialog>`,
      props: ['visualization'],
      components: { Dialog, ChangePrivacy }
    },
    { visualization }
  );
}

export function duplicateMap (visualization) {
  this.showModal(
    {
      template: `
      <Dialog v-on:close="$emit('close')">
        <DuplicateMap :visualization="visualization" v-on:close="$emit('close')"/>
      </Dialog>`,
      props: ['visualization'],
      components: { Dialog, DuplicateMap }
    },
    { visualization }
  );
}

export function changeLockState (visualization, contentType) {
  const dialogProps = { visualization, contentType };

  this.showModal(
    {
      template: `
      <Dialog v-on:close="$emit('close')">
        <LockMap :visualization="visualization" :contentType="contentType" v-on:close="$emit('close')"/>
      </Dialog>`,
      props: ['visualization', 'contentType'],
      components: { Dialog, LockMap }
    }, dialogProps
  );
}

export function changeVisualizationsLockState (visualizations, contentType) {
  const dialogProps = { visualizations, contentType };

  this.showModal(
    {
      template: `
      <Dialog v-on:close="$emit('close')">
        <LockMap :visualizations="visualizations" :contentType="contentType" v-on:close="$emit('close')"/>
      </Dialog>`,
      props: ['visualizations', 'contentType'],
      components: { Dialog, LockMap }
    }, dialogProps
  );
}

export function deleteVisualization (visualization, contentType) {
  const dialogProps = { visualization, contentType };

  this.showModal(
    {
      template: `
      <Dialog v-on:close="$emit('close')">
        <DeleteDialog :visualization="visualization" :contentType="contentType" v-on:close="$emit('close')"/>
      </Dialog>`,
      props: ['visualization', 'contentType'],
      components: { Dialog, DeleteDialog }
    }, dialogProps
  );
}

export function deleteVisualizations (visualizations, contentType) {
  const dialogProps = { visualizations, contentType };

  this.showModal(
    {
      template: `
      <Dialog v-on:close="$emit('close')">
        <DeleteDialog :visualizations="visualizations" :contentType="contentType" v-on:close="$emit('close')"/>
      </Dialog>`,
      props: ['visualizations', 'contentType'],
      components: { Dialog, DeleteDialog }
    }, dialogProps
  );
}
