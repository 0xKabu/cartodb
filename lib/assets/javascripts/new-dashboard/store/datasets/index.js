import CartoNode from 'carto-node';
import toObject from 'new-dashboard/utils/to-object';
import Filters, { defaultParams as filtersDefaultParams } from 'new-dashboard/core/filters';

const client = new CartoNode.AuthenticatedClient();

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: 'mine',
    order: 'updated_at',
    list: {},
    metadata: {},
    page: 1,
    numPages: 1
  },
  mutations: {
    setRequestError (state, error) {
      state.isFetching = false;
      state.isErrored = true;
      state.error = error;
    },

    setDatasets (state, datasets) {
      state.list = toObject(datasets.visualizations, 'id');
      state.metadata = {
        total_entries: datasets.total_entries,
        total_likes: datasets.total_likes,
        total_shared: datasets.total_shared,
        total_user_entries: datasets.total_user_entries
      };

      state.isFetching = false;
    },

    setFetchingState (state) {
      state.isFetching = true;
      state.isErrored = false;
      state.error = {};
    },

    setURLOptions (state, options) {
      state.page = options.page ? parseInt(options.page) : 1;
      state.filterType = options.filter || 'mine';
    },

    setPagination (state, page) {
      state.page = page;
      state.numPages = Math.ceil(state.metadata.total_entries / filtersDefaultParams.per_page) || 1;
    }
  },
  actions: {
    fetchDatasets (context) {
      context.commit('setFetchingState');

      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order
      };

      client.getVisualization('',
        params,
        function (err, _, data) {
          if (err) {
            context.commit('setRequestError', err);
            return;
          }
          context.commit('setDatasets', data);
          context.commit('setPagination', context.state.page);
        });
    },

    setURLOptions (context, options) {
      context.commit('setURLOptions', options);
      context.dispatch('fetchDatasets');
    }
  }
};

export default datasets;
