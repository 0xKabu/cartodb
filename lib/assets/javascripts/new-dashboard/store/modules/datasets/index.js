import Filters from 'new-dashboard/core/configuration/filters';

import * as SharedActions from '../../actions/shared';
import * as SharedMutations from '../../mutations/shared';
import * as VisualizationActions from '../../actions/visualizations';
import * as VisualizationMutations from '../../mutations/visualizations';

const DEFAULT_VALUES = {
  filter: 'mine',
  order: 'favorited,updated_at',
  orderDirection: 'desc,desc'
};

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: DEFAULT_VALUES.filter,
    order: DEFAULT_VALUES.order,
    orderDirection: DEFAULT_VALUES.orderDirection,
    list: {},
    metadata: {},
    page: 1,
    numPages: 1,
    resultsPerPage: 1
  },
  mutations: {
    setFetchingState: SharedMutations.setFetchingState,
    setFilterType: VisualizationMutations.setFilterType,
    setOrder: VisualizationMutations.setOrder(DEFAULT_VALUES),
    setPagination: SharedMutations.setPagination,
    setResultsPerPage: SharedMutations.setResultsPerPage,
    setRequestError: SharedMutations.setRequestError,
    setVisualizations: VisualizationMutations.setVisualizations,
    updateNumberLikes: VisualizationMutations.updateNumberLikes,
    updateVisualization: VisualizationMutations.updateVisualization
  },
  actions: {
    deleteLike: VisualizationActions.deleteLike,
    filter: VisualizationActions.filter,
    like: VisualizationActions.like,
    order: VisualizationActions.order,
    resetFilters: VisualizationActions.resetFilters(DEFAULT_VALUES),
    setResultsPerPage: SharedActions.setResultsPerPage,
    setURLOptions: VisualizationActions.setURLOptions,
    updateVisualization: VisualizationActions.updateVisualization,

    fetch (context) {
      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order,
        order_direction: context.state.orderDirection,
        per_page: context.state.resultsPerPage,
        with_dependent_visualizations: 10
      };

      VisualizationActions.fetchVisualizations(context, params);
    }
  }
};

export default datasets;
