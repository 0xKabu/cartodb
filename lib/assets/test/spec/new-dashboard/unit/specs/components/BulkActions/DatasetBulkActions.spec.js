import { mount } from '@vue/test-utils';
import DatasetBulkActions from 'new-dashboard/components/BulkActions/DatasetBulkActions';
import datasets from '../../fixtures/datasets';
import datasetsLocked from '../../fixtures/datasetsLocked';
import * as DialogActions from '../__mocks__/dialog-actions';

const $t = key => key;

describe('DatasetBulkActions.vue', () => {
  it('should render all bulk actions', () => {
    const datasetBulkAction = mount(DatasetBulkActions, {
      propsData: {
        areAllDatasetsSelected: false,
        selectedDatasets: []
      },
      mocks: {
        $t
      }
    });
    expect(datasetBulkAction).toMatchSnapshot();
  });

  it('should render all multiple actions for all selected "Deselect all", "Lock datasets" and "Delete datasets" actions', () => {
    const data = datasets.visualizations;
    const datasetBulkAction = mount(DatasetBulkActions, {
      propsData: {
        areAllDatasetsSelected: true,
        selectedDatasets: data
      },
      mocks: {
        $t
      }
    });
    expect(datasetBulkAction).toMatchSnapshot();
  });

  it('should render all single actions: "Select all", "Create map", "Change privacy", "Duplicate dataset", "Lock dataset", "Delete dataset"', () => {
    const data = datasets.visualizations.slice(0, 1);
    const datasetBulkAction = mount(DatasetBulkActions, {
      propsData: {
        areAllDatasetsSelected: false,
        selectedDatasets: data
      },
      mocks: {
        $t
      }
    });
    expect(datasetBulkAction).toMatchSnapshot();
  });

  it('should render all multiple actions: "Select all", "Deselect all", "Lock datasets", "Delete datasets"', () => {
    const data = datasets.visualizations;
    const datasetBulkAction = mount(DatasetBulkActions, {
      propsData: {
        areAllDatasetsSelected: false,
        selectedDatasets: data
      },
      mocks: {
        $t
      }
    });
    expect(datasetBulkAction).toMatchSnapshot();
  });

  it('should render single locked action: "Unlock dataset"', () => {
    const data = datasetsLocked.visualizations;
    const datasetBulkAction = mount(DatasetBulkActions, {
      propsData: {
        areAllDatasetsSelected: false,
        selectedDatasets: data
      },
      mocks: {
        $t
      }
    });
    expect(datasetBulkAction).toMatchSnapshot();
  });

  it('should render multiple locked action: "Unlock datasets"', () => {
    const data = datasetsLocked.visualizations.slice(0, 2);
    const datasetBulkAction = mount(DatasetBulkActions, {
      propsData: {
        areAllDatasetsSelected: false,
        selectedDatasets: data
      },
      mocks: {
        $t
      }
    });
    expect(datasetBulkAction).toMatchSnapshot();
  });

  describe('computed', () => {
    it('should return "multiple" action name when more than one dataset is selected', () => {
      const data = datasets.visualizations;
      const datasetBulkAction = mount(DatasetBulkActions, {
        propsData: {
          selectedDatasets: data
        },
        mocks: {
          $t
        }
      });

      const result = datasetBulkAction.vm.actionMode;
      expect(result).toBe('multiple');
    });

    it('should return "single" action name when there is only a dataset selected', () => {
      const data = datasets.visualizations.slice(0, 1);
      const datasetBulkAction = mount(DatasetBulkActions, {
        propsData: {
          selectedDatasets: data
        },
        mocks: {
          $t
        }
      });
      const result = datasetBulkAction.vm.actionMode;
      expect(result).toBe('single');
    });

    it('should return "lock" action name when there is only a locked dataset selected', () => {
      const data = datasetsLocked.visualizations.slice(0, 1);
      const datasetBulkAction = mount(DatasetBulkActions, {
        propsData: {
          selectedDatasets: data
        },
        mocks: {
          $t
        }
      });
      const result = datasetBulkAction.vm.actionMode;
      expect(result).toBe('lock');
    });

    it('should return "multipleLock" action name when there are more than one locked dataset selected', () => {
      const data = datasetsLocked.visualizations;
      const datasetBulkAction = mount(DatasetBulkActions, {
        propsData: {
          selectedDatasets: data
        },
        mocks: {
          $t
        }
      });
      const result = datasetBulkAction.vm.actionMode;
      expect(result).toBe('multipleLock');
    });
  });

  describe('methods', () => {
    let $modal;
    let data;
    let datasetBulkAction;
    beforeEach(function () {
      $modal = {
        show: jest.fn()
      };
      data = datasets.visualizations;
      datasetBulkAction = mount(DatasetBulkActions, {
        propsData: {
          selectedDatasets: data
        },
        mocks: {
          $modal,
          $t,
          backboneViews: {
            backgroundPollingView: {
              getBackgroundPollingView () {
                return {};
              }
            }
          }
        }
      });
    });

    it('should call the function to show modal', () => {
      datasetBulkAction.vm.showModal();
      expect($modal.show).toHaveBeenCalled();
    });

    it('should call the function selectAll', () => {
      datasetBulkAction.vm.selectAll();
      expect(datasetBulkAction.emitted('selectAll').length).toBe(1);
    });

    it('should call the function deselectAll', () => {
      datasetBulkAction.vm.deselectAll();
      expect(datasetBulkAction.emitted('deselectAll').length).toBe(1);
    });

    it('should open create map', () => {
      datasetBulkAction.vm.createMap();
      expect(DialogActions.createMap).toHaveBeenCalled();
    });

    it('should open change privacy modal', () => {
      datasetBulkAction.vm.changePrivacy();
      expect(DialogActions.changePrivacy).toHaveBeenCalled();
    });

    it('should open duplicate dataset modal', () => {
      datasetBulkAction.vm.duplicateDataset();
      expect(DialogActions.duplicateDataset).toHaveBeenCalled();
    });

    it('should open change visualizations lock state dialog via unlockDatasets funcion', () => {
      datasetBulkAction.vm.unlockDatasets();
      expect(DialogActions.changeVisualizationsLockState).toHaveBeenCalled();
    });

    it('should open change visualizations lock state dialog via lockDatasets funcion', () => {
      datasetBulkAction.vm.lockDatasets();
      expect(DialogActions.changeVisualizationsLockState).toHaveBeenCalled();
    });

    it('should open change lock state dialog via lockDataset function', () => {
      datasetBulkAction.vm.lockDataset();
      expect(DialogActions.changeLockState).toHaveBeenCalled();
    });

    it('should open change lock state dialog via unlockDataset function', () => {
      datasetBulkAction.vm.unlockDataset();
      expect(DialogActions.changeLockState).toHaveBeenCalled();
    });

    it('should open delete visualization dialog', () => {
      datasetBulkAction.vm.deleteDataset();
      expect(DialogActions.deleteVisualization).toHaveBeenCalled();
    });

    it('should open delete visualizations dialog', () => {
      datasetBulkAction.vm.deleteDatasets();
      expect(DialogActions.deleteVisualizations).toHaveBeenCalled();
    });
  });
});
