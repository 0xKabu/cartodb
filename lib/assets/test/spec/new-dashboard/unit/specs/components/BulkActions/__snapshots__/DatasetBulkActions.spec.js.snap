// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetBulkActions.vue should render all bulk actions 1`] = `
<section class="bulk-actions"><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.selectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.createMap
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.changeDatasetPrivacy
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.duplicateDataset
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDataset
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDataset
  </button></section>
`;

exports[`DatasetBulkActions.vue should render all multiple actions for all selected "Create all", Deselect all", "Lock datasets" and "Delete datasets" actions 1`] = `
<section class="bulk-actions">
  <!----><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.createMap
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.deselectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDatasets
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDatasets
  </button></section>
`;

exports[`DatasetBulkActions.vue should render all multiple actions for all selected "Deselect all", "Lock datasets" and "Delete datasets" actions 1`] = `
<section class="bulk-actions">
  <!----><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.deselectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDatasets
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDatasets
  </button></section>
`;

exports[`DatasetBulkActions.vue should render all multiple actions for all selected "Select all", "Create map", Deselect all", "Lock datasets" and "Delete datasets" actions 1`] = `
<section class="bulk-actions">
  <!----><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.createMap
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.deselectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDatasets
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDatasets
  </button></section>
`;

exports[`DatasetBulkActions.vue should render all multiple actions: "Select all", "Create map", Deselect all", "Lock datasets", "Delete datasets" 1`] = `
<section class="bulk-actions"><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.selectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.createMap
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.deselectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDatasets
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDatasets
  </button></section>
`;

exports[`DatasetBulkActions.vue should render all multiple actions: "Select all", "Deselect all", "Lock datasets", "Delete datasets" 1`] = `
<section class="bulk-actions"><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.selectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.deselectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDatasets
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDatasets
  </button></section>
`;

exports[`DatasetBulkActions.vue should render all single actions: "Select all", "Create map", "Change privacy", "Duplicate dataset", "Lock dataset", "Delete dataset" 1`] = `
<section class="bulk-actions"><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.selectAllDatasets
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.createMap
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.changeDatasetPrivacy
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.duplicateDataset
  </button><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.lockDataset
  </button><button class="bulk-actions__button button button--ghost is-txtAlert">
    BulkActions.datasets.deleteDataset
  </button></section>
`;

exports[`DatasetBulkActions.vue should render multiple locked action: "Unlock datasets" 1`] = `
<section class="bulk-actions"><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.unlockDatasets
  </button></section>
`;

exports[`DatasetBulkActions.vue should render single locked action: "Unlock dataset" 1`] = `
<section class="bulk-actions"><button class="bulk-actions__button button button--ghost">
    BulkActions.datasets.unlockDatasets
  </button></section>
`;
